source/extensions/common/wasm/ext/verify_signature.proto

```diff
@@ -2,14 +2,17 @@ syntax = "proto3";
 
 package envoy.source.extensions.common.wasm;
 
+// Arguments for verifying a signature with a public key.
 message VerifySignatureArguments {
-  string hash_function = 1;
-  bytes public_key = 2;
-  bytes signature = 3;
-  bytes text = 4;
-};
+  string hash_function = 1;  // Hash function used (e.g., "sha256")
+  bytes public_key = 2;      // Public key in DER format
+  bytes signature = 3;       // Signature to verify
+  bytes text = 4;            // Original data that was signed
+  string public_key_pem = 5; // Public key in PEM format (mutually exclusive with public_key)
```

Comment:
This would be a nicer interface if you put the two public_key options in a oneof. Also, name `public_key` to `public_key_der` for clarity.